src/main/java/com/treasuredata/android/cdp/FetchUserSegmentsCallback.java

```diff
@@ -0,0 +1,19 @@
+package com.treasuredata.android.cdp;
+
+import java.util.List;
+
+public interface FetchUserSegmentsCallback {
```

Comment:
IMHO, this should be JSONLookupCallback. With FetchUserSegmentsCallback extends JSONLookupCallback. Overridden onSuccess and onError will be perfect place to convert JSON object to ProfileImpl POJO.

Reply:
FetchUserSegmentsCallback should not be a sub-case of JSONLookupCallback, it should be the other way around. Plus this is a public interface, and user codes should not concern about these.
FetchUserSegmentsCallback just means this is a callback interface for FetchUserSegments operation, nothing more.